### DISCUSSION TITLE
weberp.xmlrpc_StockAdjustment API throws error 500 when called

### DISCUSSION THREAD

**Discussion by minhnguyen-ew** [timschofield/webERP#173] [Problems / Bugs] [answered]
I am using Java for communicating with the WebERP XML-RPC API. Call the above API and I receive the error 500 with this log.

`[Thu Mar 29 09:49:17.788294 2018] [:error] [pid 30695] [client 115.79.192.118:41532] PHP Fatal error:  Call to undefined function GetNextTransactionNo() in /var/www/html/jio/api/api_stock.php on line 809`

I am using version 4.14.1 downloaded from SourceForge.

**Comment by andrewcouling** [comment]
I recommend  that you post your query on the WebERP support forum here: http://www.weberp.org/forum/

**Comment by timschofield** [comment]
Try changing the GetNextTransactionNo references to GetNextTransNo in api_stock.php.

@andrewcouling I am told you can sometimes wait days to get your account on the forum.

Tim

**Comment by timschofield** [comment, marked as answer]
I have fixed this issue, you need to download the following files from here:
        api/api_debtortransactions.php
        api/api_stock.php
        api/api_workorders.php
Do not download the whole repo as the xmlrpc library is broken.

Tim

**Comment by minhnguyen-ew** [comment]
Work flawlessly now, thanks.

**Comment by dalers** [comment]
@timschofield I'm confused by

> Do not download the whole repo as the xmlrpc library is broken.

Does this require development to resolve? (I've been reviewing old Discussions and Issues...).

**Comment by timschofield** [comment]
I honestly don't remember what the issue was Dale, I have to check with my wife what I had for breakfast, let alone remember 6 1/2 years ago :)

I have updated the xmlrpc library recently so it should be OK now.
